### ISSUE TITLE
[Bug Reporting] Multicolumn uniqueness check is not working.

### ISSUE THREAD

**Issue by heyuan0403** [fivetran/great_expectations#1807]
**Describe the bug**
`expect_multicolumn_values_to_be_unique` is not working as expected. As we can manually see in the file that there are two identical rows across below columns for statement id value `2085500` 
```
statement_id
statement_closing_date
card_reference_number
```
but the GE expectation validation result says `"success": true`  when I run `batch.expect_multicolumn_values_to_be_unique(column_list = ['statement_closing_date', 'card_reference_number'])`

**To Reproduce**
I was running all this locally but here are the code blocks that I have run in notebook
```
*CELL:* context = ge.data_context.DataContext()

*CELL:* context.list_expectation_suite_names()
*OUTPUT:* ['petal_redshift.default.i2c_statement_account_details.i2c_statement_account_details_data_quality']

*CELL:* expectation_suite_name =
"petal_redshift.default.i2c_statement_account_details.i2c_statement_account_details_data_quality"

*CELL:* [datasource['name'] for datasource in context.list_datasources() if
datasource['class_name'] == 'PandasDatasource']
datasource_name = 'petal_s3'
batch_kwargs = {'path': "non_pii_stmt.csv", 'datasource': datasource_name}

*CELL:* batch = context.get_batch(batch_kwargs, expectation_suite_name)

*CELL:* batch.expect_multicolumn_values_to_be_unique(column_list = ['statement_closing_date', 'card_reference_number'])

*OUTPUT:*
 {
  "meta": {},
  "result": {
    "element_count": 40,
    "missing_count": 0,
    "missing_percent": 0.0,
    "unexpected_count": 0,
    "unexpected_percent": 0.0,
    "unexpected_percent_nonmissing": 0.0,
    "partial_unexpected_list": []
  },
  "success": true,
  "exception_info": null
}
```

**Expected behavior**

The expectation result from `expect_multicolumn_values_to_be_unique` should be `"success": true` 

**Environment (please complete the following information):**
 - OS: MacOS Catalina 10.15.6
 - GE Version:  0.10.12

**Additional context**
I have attached the CSV, expectation suite, and yml file below in the zip. Please let me know if you are able to reproduce.
[for_ge.zip](https://github.com/great-expectations/great_expectations/files/5075743/for_ge.zip)

**Comment by jliew** [comment]
You're wanting row-level uniqueness, whereas expect_multicolumn_values_to_be_unique checks for uniqueness across column-level axis.  You can see how it calls `.nunique(axis=1)` here: https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/dataset/pandas_dataset.py#L1788

I had the same misunderstanding too on initial reading, maybe the description for this expectation could be made more explicit?

**Comment by anthonyburdi** [comment]
@jcampbell @eugmandel - Eugene and I just chatted about this and we are thinking it may be a good idea to rename the existing `expect_multicolumn_values_to_be_unique` to something more like `expect_row_values_to_be_unique`. I also started a PR (#1822) with a version of `expect_multicolumn_values_to_be_unique_together` for the pandas dataset. FYI another user asked for this for a spark df as well this week in GE support.

**Comment by heyuan0403** [comment]
Got it, thanks for looking into this, I'm going to close this thread then

**Comment by anthonyburdi** [comment]
No problem! We would like to leave this issue open until it's resolved, but thank you again for bringing it up!

**Comment by tscottcoombes1** [comment]
spark expectation:

`@MetaSparkDFDataset.multicolumn_map_expectation
    def expect_composite_column_values_to_be_unique(
        self,
        column_list,  # pyspark.sql.DataFrame
    ):
        # Might want to throw an exception if only 1 column is passed
        column_names = column_list.schema.names[:]
        
        return column_list.withColumn(
            "__success", count(lit(1)).over(Window.partitionBy(struct(*column_names))) <= 1
        )`

**Comment by anthonyburdi** [comment]
Hi @heyuan0403 , @jliew ! We have merged PR #1822 which hopefully clarifies things! It will be in the next Great Expectations release. 

A quick summary of 1822: We have added a new expectation called `expect_compound_columns_to_be_unique` which is a bit more like what you are discussing (currently for the Pandas backend). Here is the docstring:
```
        Expect that the columns are unique together, e.g. a multi-column primary key
        Note that all instances of any duplicates are considered failed
        E.g.
        A B C
        1 1 2 Fail
        1 2 3 Pass
        1 1 2 Fail
        2 2 2 Pass
        3 2 3 Pass
```
We have deprecated `expect_multicolumn_values_to_be_unique` and replaced it with `expect_select_column_values_to_be_unique_within_record` which is hopefully a bit more clear based on how that expectation worked.

@tscottcoombes1 - thank you so much for the comment! We have added issue #1920 for spark support and #1921 for sqlalchemy. If you are interested to turn your comment into a PR, we love community contributions!
